tests/tests_common.py

```diff
@@ -699,6 +699,87 @@ def test_base_data_type(self):
         self.assertEqual(ua_utils.get_base_data_type(d), self.opc.get_node(ua.ObjectIds.Structure))
         self.assertEqual(ua_utils.data_type_to_variant_type(d), ua.VariantType.ExtensionObject)
 
-
-
-
+    def test_ua_method_unique(self):
+        def unique_response(parent):
+            return 42
+
+        decorated_unique_response = uamethod(unique_response)
+        response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+
+        self.assertEqual(len(response), 1)
+        self.assertEqual(type(response[0]), ua.Variant)
+
+    def test_ua_method_multiple(self):
+        def unique_response(parent):
+            return 42, 49
+
+        decorated_unique_response = uamethod(unique_response)
+        response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+
+        self.assertEqual(len(response), 2)
+        self.assertEqual(type(response[0]), ua.Variant)
+
+    def test_ua_method_complex(self):
+        register_extension_object(KeyValuePair)
+
+        def unique_response(parent):
+            return KeyValuePair("Key", "Value")
+
+        decorated_unique_response = uamethod(unique_response)
+        response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+
+        self.assertEqual(len(response), 1)
+        self.assertEqual(type(response[0]), ua.Variant)
+
+    def test_ua_method_complex_multiple(self):
+        register_extension_object(KeyValuePair)
+
+        def unique_response(parent):
+            return [KeyValuePair("Key", "Value"),
+                    KeyValuePair("Hello", "World")]
+
+        decorated_unique_response = uamethod(unique_response)
+        response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+
+        self.assertEqual(len(response), 2)
+        self.assertEqual(type(response[0]), ua.Variant)
+
+    # TODO: Fix the test
+    # def test_decode_custom_object(self):
+    #     register_extension_object(KeyValuePair)
+    #
+    #     def unique_response(parent):
+    #         return KeyValuePair("Key", "Value")
+    #
+    #     decorated_unique_response = uamethod(unique_response)
+    #     response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+    #
+    #     self.assertEqual(len(response), 1)
+    #     self.assertEqual(type(response[0]), ua.Variant)
+    #
+    #     key_value = uabin.unpack_uatype_array(response[0].VariantType, io.BytesIO(response[0].to_binary()))
+    #     self.assertEqual(key_value.key, "Key")
+    #     self.assertEqual(key_value.value, "Value")
```

Comment:
I got this error:

`======================================================================
ERROR: test_decode_custom_object (tests_server.TestServer)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Work\doc\Open_SCS\opcua-kev\python-opcua\tests\tests_common.py", line 762, in test_decode_custom_object
    key_value = uabin.unpack_uatype_array(response[0].VariantType, io.BytesIO(response[0].to_binary()))
  File ".\opcua\ua\ua_binary.py", line 305, in unpack_uatype_array
    return [unpack_uatype(vtype, data) for _ in range(length)]
  File ".\opcua\ua\ua_binary.py", line 305, in <listcomp>
    return [unpack_uatype(vtype, data) for _ in range(length)]
  File ".\opcua\ua\ua_binary.py", line 286, in unpack_uatype
    return extensionobject_from_binary(data)
  File ".\opcua\ua\uaprotocol_auto.py", line 16300, in extensionobject_from_binary
    TypeId = NodeId.from_binary(data)
  File ".\opcua\ua\uatypes.py", line 410, in from_binary
    nid.NodeIdType = NodeIdType(encoding & 0b00111111)
  File "C:\Python3\lib\enum.py", line 241, in __call__
    return cls.__new__(cls, value)
  File "C:\Python3\lib\enum.py", line 476, in __new__
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 14 is not a valid NodeIdType`

It looks like a configuration error but I'm not sure about how to fix it.